src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -732,23 +730,37 @@ class AnonymousFunctions
                 Assert.Equal(1, diagnostics.Count())
                 Assert.Equal(document.Id, diagnostics.First().DocumentId)
 
-                ' Verify compilation diagnostics are reported with correct location info when asked for project diagnostics.
-                Dim projectDiagnostics = diagnosticService.GetProjectDiagnosticsForIdsAsync(project.Solution, project.Id).WaitAndGetResult(CancellationToken.None)
-                Assert.Equal(2, projectDiagnostics.Count())
-
-                Dim noLocationDiagnostic = projectDiagnostics.First()
-                Assert.Equal(CompilationEndedAnalyzer.Descriptor.Id, noLocationDiagnostic.Id)
-                Assert.Equal(False, noLocationDiagnostic.HasTextSpan)
-
-                Dim withDocumentLocationDiagnostic = projectDiagnostics.Last()
-                Assert.Equal(CompilationEndedAnalyzer.Descriptor.Id, withDocumentLocationDiagnostic.Id)
-                Assert.Equal(True, withDocumentLocationDiagnostic.HasTextSpan)
-                Assert.NotNull(withDocumentLocationDiagnostic.DocumentId)
-                Dim diagnosticDocument = project.GetDocument(withDocumentLocationDiagnostic.DocumentId)
-                Dim tree = diagnosticDocument.GetSyntaxTreeAsync().Result
-                Dim actualLocation = withDocumentLocationDiagnostic.ToDiagnosticAsync(project, CancellationToken.None).Result.Location
-                Dim expectedLocation = document.GetSyntaxRootAsync().Result.GetLocation
-                Assert.Equal(expectedLocation, actualLocation)
+                If options.GetOption(InternalDiagnosticsOptions.UseDiagnosticEngineV2) Then
+                    ' Verify compilation diagnostics are reported with correct location info when asked for project diagnostics.
+                    Dim projectDiagnostics = diagnosticService.GetProjectDiagnosticsForIdsAsync(project.Solution, project.Id).WaitAndGetResult(CancellationToken.None)
+                    Assert.Equal(1, projectDiagnostics.Count())
+
+                    Dim noLocationDiagnostic = projectDiagnostics.First()
+                    Assert.Equal(CompilationEndedAnalyzer.Descriptor.Id, noLocationDiagnostic.Id)
+                    Assert.Equal(False, noLocationDiagnostic.HasTextSpan)
+                Else
+                    ' REVIEW: GetProjectDiagnosticsForIdsAsync is for project diagnostics with no location. not sure why in v1, this API returns
+                    '         diagnostic with location?
+
+                    ' Verify compilation diagnostics are reported with correct location info when asked for project diagnostics.
+                    Dim projectDiagnostics = diagnosticService.GetProjectDiagnosticsForIdsAsync(project.Solution, project.Id).WaitAndGetResult(CancellationToken.None)
```

Comment:
@heejaechang Test is attempting to verify diagnostics with location reported from compilation end actions - it should probably be just calling GetDiagnosticsAsync and then you can probably avoid the If-Else here?